### ISSUE TITLE
PEP723 Support 

### ISSUE THREAD

**Issue by daylinmorgan** [daylinmorgan/viv#14]
PEP723 is a direct competitor to PEP722. 
Based on online discussions it seems like there might be more preference for it over PEP722. 

Implementing this for python <3.11 is tricky since I don't want to expand `viv` to more than one file or add a third-party dependency.
I see two options:

- use a zipapp with `tomli`
- embed a fully vendored albeit slightly modified `tomli` within `viv.py`

I've implemented the second option in the [pep723](https://github.com/daylinmorgan/viv/tree/pep723) branch.
Examples:
```sh
python3 <(curl -fsSL https://raw.githubusercontent.com/daylinmorgan/viv/pep723/src/viv/viv.py) run -s https://raw.githubusercontent.com/daylinmorgan/viv/pep723/examples/pep723.py
```

Installation options for now:
```sh
python3 <(curl -fsSL viv.dayl.in/viv.py) manage install --ref pep723
viv manage update --ref pep723
```

**Comment by daylinmorgan** [comment]
As I expected PEP723 was the winner. I'll likely remove [PEP722 support](#5) in favor of supporting PEP723.

**Comment by daylinmorgan** [comment]
[Consensus](https://discuss.python.org/t/how-would-you-like-to-declare-runtime-dependencies-and-python-requirements-for-pep-723/40418/70?u=daylinmorgan) is getting closer on a pep723 syntax.

Something like:

```python
# /// script
# requires-python = ">=3.12"
# dependencies = ["httpx"]
#
# [tool.magic]
# style = "interesting"
#
# [tool.linter]
# strict = true
# ///
```

Current implementation in [pep723 branch](https://github.com/daylinmorgan/viv/tree/pep723) should be updated prior to official release and subsequent deprecation of pep722 syntax.

**Comment by daylinmorgan** [comment]
Released in [v2024.1004](https://github.com/daylinmorgan/viv/tree/v2024.1004)
